src/features/game/components/modal/components/BuyGems.tsx

```diff
@@ -120,6 +127,88 @@ export const BuyGems: React.FC<Props> = ({
     );
   }
 
+  if (!!price && showFlowerConfirm) {
+    const flowerPrice =
+      gameService.getSnapshot().context.prices.sfl?.usd ?? 0.0;
```

Comment:
Is there a reason to mix these accessors :
`gameService.getSnapshot()`
`gameService.state.context.state.balance`

Neither of these subscribe to changes. The `useSelector` pattern which is used above is a better one to follow. Unless there is a reason for using this "point in time" value. 

Mixing the ways we access data *can* lead to unexpected behaviour. I guess so long as you use these with intention its fine.

Reply:
Thanks mate. I will patch those up.

I often use the gameService when re-renders are not necessary - I will make them consistent